paint/PAINT_ButtonFunctions.cpp

```diff
@@ -3,29 +3,69 @@
 #include <iostream>
 #include "PAINT_Utils.h"
 #include "PAINT_DrawWindow.h"
+#include "PAINT_Brush.h"
+#include "WIN_ToggleButton.h"
 
 using namespace win;
 using namespace paint;
 
-//void myAction(win::Button* button)
-//{
-//	std::cout << "Im doing an action yay! \n";
-//}
-//
-//void paint::foo()
-//{
-//	Button b;
-//	b.action = myAction;
-//}
-
 void paint::myAction(win::Button* button)
 {
 	std::cout << "Im doing an action yay! \n";
 }
 
-void paint::toggleDraw(win::Button* button)
+void paint::toggleDraw(win::ToggleButton* button)
+{
+	auto dw = utils::findDrawWindow(button);
+	dw->toggleDrawTool(button);
+}
+
+void paint::setBrushThickness0(win::ToggleButton* button)
+{
+	auto dw = utils::findDrawWindow(button);
+	const auto activetool = dw->getActiveTool();
+	if (activetool)
+	{
+		const auto activebrush = activetool->getActiveBrush();
+		if (activebrush) {
+			activebrush->setThickness(0);
+		}
+	}
+}
+
+void paint::setBrushThickness1(win::ToggleButton* button)
```

Comment:
These two functions are so similar they should probably call another function

`static void setBrushThickness(Brush * b, int thickness);`

Classic case of DRY :-)

Reply:
done